sns_ik_lib/matlab/snsIk_acc_rr_mt.m

```diff
@@ -0,0 +1,254 @@
+                Wistar = Wi;
+                ddqNulliStar = ddqNulli;
+                PiBarStar = PiBar;
+                PiHatStar = PiHat;
+            end
+
+            % saturate the most critical joint
+            Wi(JntIdx, JntIdx) = 0;
+            ddqNulli(JntIdx) = min(max(ddqLow(JntIdx), ddqi(JntIdx)), ddqUpp(JntIdx));
+
+            % update projection matrices
+            PiBar = (I - pinv((I - Wi)*PiPrev, tol))*PiPrev;
+            PiHat = (I - pinv(Ji*PiBar, tol)*Ji)*pinv((I - Wi)*PiPrev, tol);
+
+            % if rank is below task dimension, terminate the loop and output
+            % the current best solution
+            if (rank(Ji*PiBar) < nTask)
+                si = siStar;
+                Wi = Wistar;
+                ddqNulli = ddqNulliStar;
+                PiBar = PiBarStar;
+                PiHat = PiHatStar;
+
+                if (drop_correction_term)
+                    ddqi = ddqiPrev + pinv(Ji*PiBar, tol) * (si*ddxGoali - dJdqi) + ...
+                        PiHat*(ddqNulli - ddqiPrev);
+
+                    limitExceeded = false;
+
+                    % if the solution for lower priority tasks violates the limits, ignore the solution
+                    if (iTask > 1 && (any(ddqi > ddqUpp + tol) || any(ddqi < ddqLow - tol)))
+                        si = 0;
+                        Wi = zeros(nJnt, nJnt);
+                        ddqi = ddqiPrev;
+                    end
+                else
+                    ddqi = ddqiPrev + pinv(Ji*PiBar, tol) * (si*ddxGoali - dJdqi - Ji*ddqiPrev) + ...
+                        PiHat*(ddqNulli - ddqiPrev);
+
+                    % if the solution violates the limits, drop the
+                    % correction term and run it again
+                    if (any(ddqi > ddqUpp + tol) || any(ddqi < ddqLow - tol))
+                        drop_correction_term = true;
+                        cntLoop = 0;
+                        Wi = eye(nJnt);
+                        ddqNulli = zeros(nJnt, 1);
+                        PiBar = PiPrev;
+                        si = 1.0;
+                        siStar = 0.0;
+                    else
+                        limitExceeded = false;
+                    end
+                end
+            end
+
+        else % if the current task is infeasible
+            si = 0;
+            Wi = zeros(nJnt, nJnt);
+            ddqi = ddqiPrev;
+            limitExceeded = false;
+        end
+
+        cntLoop = cntLoop + 1;
+    end
+
+    % only if the current task is feasible, update nullspace projection
+    if (si > 0)
+        Pi = PiPrev - pinv(Ji*PiPrev, tol)*(Ji*PiPrev);
+    end
+
+    % store data
+    WData{iTask} = Wi;
+    sData(iTask) = si;
+    ddqData{iTask} = ddqi;
+    ddqNullData{iTask} = ddqNulli;
+end
+
+%-- end of algorithm
+ddq = ddqi;
+
+end
+
+%~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~%
+
+function taskScale = FindScaleFactor(low, upp, a)
```

Comment:
It looks like this function is included in about eight different files. I propose that you put it in a stand-along file and then remove it from each of the solvers. Please also provide documentation for it.